src/stamps.ts

```diff
@@ -307,17 +307,17 @@ export class StampsManager {
   async start(config: StampsConfig): Promise<void> {
     // Hardcoded stamp mode
     if (config.mode === 'hardcoded') this.stamp = config.stamp
-    // Autobuy mode
+    // Autobuy or ExtendsTTL mode
     else {
+      if (config.mode === 'autobuy') {
+        this.refreshStamps = async () => this.refreshStampsAutobuy(config, new BeeDebug(config.beeDebugApiUrl))
+      } else {
+        this.refreshStamps = async () => this.refreshStampsExtends(config, new BeeDebug(config.beeDebugApiUrl))
+      }
       this.stop()
+      await this.refreshStamps()
```

Comment:
I assume that you create the new instance property `refreshStamps` only so you could assign it inside the `if` scope and then, later on, call it and set it in the interval right? That is IMHO not a good solution as you don't really need the `refreshStamps` anywhere else in the class and it pollutes unnecessary the instance space.

Usually, this problem is solved by using `let` instead of `const`, so you could do something along these lines:
```ts
let refreshStamps: () => Promise<void>

if(...){
   refreshStamps = ...
} else {
   refreshStamps = ...
}

refreshStamps()
```

Reply:
Yes, I'm not much of a fan of `let` variables. But yes, you are right, `refreshStamps` is not like the other global variables that are used in several methods.